hermes-consumers/src/test/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSenderTest.java

```diff
@@ -287,51 +224,6 @@ public void shouldBackoffRetriesWhenEndpointFails() throws InterruptedException
         verifyErrorHandlerHandleFailed(message, subscriptionWithBackoff, 1 + executionTime / senderBackoffTime);
     }
 
-    @Test
-    public void shouldNotBackoffRetriesOnRetryAfter() throws InterruptedException {
-        // given
-        int retrySeconds = 1;
-        Message message = message();
-        doReturn(backoff(retrySeconds)).doReturn(success()).when(messageSender).send(message);
-
-        // when
-        sender.sendAsync(message);
-
-        // then
-        verifyRateLimiterFailedSendingCountedTimes(0);
-        verifyRateLimiterSuccessfulSendingCountedTimes(2);
-        verifySemaphoreReleased();
-    }
-
-    @Test
-    public void shouldBackoffRetriesOnServiceUnavailableWithoutRetryAfter() throws InterruptedException {
-        // given
-        Message message = message();
-        doReturn(failure(SERVICE_UNAVAILABLE.code())).doReturn(success()).when(messageSender).send(message);
-
-        // when
-        sender.sendAsync(message);
-
-        // then
-        verifyRateLimiterFailedSendingCountedTimes(1);
-        verifyRateLimiterSuccessfulSendingCountedTimes(1);
-        verifySemaphoreReleased();
-    }
-
-    @Test
-    public void shouldBackoffRetriesOnTooManyRequestsWithoutRetryAfter() throws InterruptedException {
```

Comment:
This test was broken. From it's name I would assume the following would happen

```
response: TOO_MANY_REQUESTS
registerFailedSending()
response: OK
registerSuccesfulSending()
```

but actually it is

```
response: TOO_MANY_REQUESTS
registerSuccesfulSending()
```

request is not even retried because TMR is considered a client error and subscription under test doesn't have RetryClientErrors policy

Reply:
ok, so retries for 4xx depend on `Retry on 4xx status` flag for subscription. If the flag is turned off, the 4xx is treated as success, so it does not count to output rate calculator and does not block messages sending

Reply:
yup, the test has concerns bad type of subscription: it should be subscription with retries enabled on 4xx